src/lib.rs

```diff
@@ -1076,6 +1140,39 @@ where
             let txid = maybe_await!(wallet.broadcast(&tx))?;
             Ok(json!({ "txid": txid }))
         }
+        #[cfg(feature = "reserves")]
+        ProduceProof { msg } => {
+            let mut psbt = maybe_await!(wallet.create_proof(&msg))?;
+
+            let _finalized = wallet.sign(
+                &mut psbt,
+                SignOptions {
+                    trust_witness_utxo: true,
+                    ..Default::default()
+                },
+            )?;
+
+            let psbt_ser = serialize(&psbt);
+            let psbt_b64 = base64::encode(&psbt_ser);
+
+            Ok(json!({ "psbt": psbt , "psbt_base64" : psbt_b64}))
+        }
+        #[cfg(feature = "reserves")]
+        VerifyProof {
+            psbt,
+            msg,
+            confirmations,
+        } => {
+            let psbt = base64::decode(&psbt).unwrap();
+            let psbt: PartiallySignedTransaction = deserialize(&psbt).unwrap();
+            let current_height = wallet.client().get_height()?;
+            let max_confirmation_height = if confirmations == 0 { None } else { Some(current_height - confirmations) };
+
+            maybe_await!(wallet.sync(log_progress(), None))?;
```

Comment:
If the user does a manual sync before calling the `verify_proof` command will it work? If so then I think that's ok and you don't need to add it in the code here.. Other command like `list_transactions`  etc. also expect the user to sync their wallet first.

Reply:
No, user sync is not enough. You can verify that by commenting out this sync line, and then running the proof_of_reserves_roundtrip.sh.txt file that I attached to the comment above.

Reply:
The interesting thing is that with user sync, but without this additional line wallet.list_unspent() works perfectly fine, but wallet.get_tx() does not.

Reply:
Ah, I found the culprit. I had to rearrange the commands in proof_of_reserves_roundtrip.sh.txt so that the sync for the public descriptor wallet was performed right before their use. Before there were some calls with the private descriptor wallet in between. Somehow I assumed the default wallet name from descriptor (https://github.com/bitcoindevkit/bdk-cli/pull/57) was already in use.
With that I could remove the additional sync again.